src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -183,9 +183,10 @@ void setupConnection(final Connection connection, final long connectionTimeout)
       connection.setAutoCommit(isAutoCommit);
       connection.setReadOnly(isReadOnly);
 
-      final int defaultLevel = connection.getTransactionIsolation();
-      transactionIsolation = (transactionIsolation < 0 ? defaultLevel : transactionIsolation);
-      if (transactionIsolation != defaultLevel) {
+      if (transactionIsolation == -1) {
+         transactionIsolation = connection.getTransactionIsolation();
+      }
+      else if (transactionIsolation != Connection.TRANSACTION_NONE) {
```

Comment:
...but even if the user specifies ``Connection.TRANSACTION_NONE`` we would need to call the setter.

Additionally, this code is *less efficient* because the code on the left only calls ``setTransactionIsolation()`` if the user specified level is different than the default.  *This* code will always call ``setTransactionIsolation()``, even if the user specified level is the same as the connection default.